snekbox/nsjail.py

```diff
@@ -163,37 +220,61 @@ def python3(
                 "--",
                 self.config.exec_bin.path,
                 *self.config.exec_bin.arg,
-                *py_args,
-                code,
-            )
+                # Filter out empty strings at start of py_args
+                # (causes issues with python cli)
+                *iter_lstrip(py_args),
+            ]
+
+            # Write provided files if any
+            for file in files:
+                try:
+                    file.save_to(fs.home)
+                    log.info(f"Created file at {(fs.home / file.path)!r}.")
+                except OSError as e:
+                    log.info(f"Failed to create file at {(fs.home / file.path)!r}.", exc_info=e)
+                    return EvalResult(
+                        args, None, f"{e.__class__.__name__}: Failed to create file '{file.path}'."
+                    )
 
             msg = "Executing code..."
             if DEBUG:
-                msg = f"{msg[:-3]}:\n{textwrap.indent(code, '    ')}\nWith the arguments {args}."
+                msg = f"{msg[:-3]} with the arguments {args}."
             log.info(msg)
 
             try:
                 nsjail = subprocess.Popen(
                     args, stdout=subprocess.PIPE, stderr=subprocess.STDOUT, text=True
                 )
             except ValueError:
-                return CompletedProcess(args, None, "ValueError: embedded null byte", None)
+                return EvalResult(args, None, "ValueError: embedded null byte")
 
             try:
                 output = self._consume_stdout(nsjail)
             except UnicodeDecodeError:
-                return CompletedProcess(
-                    args,
-                    None,
-                    "UnicodeDecodeError: invalid Unicode in output pipe",
-                    None,
-                )
+                return EvalResult(args, None, "UnicodeDecodeError: invalid Unicode in output pipe")
 
             # When you send signal `N` to a subprocess to terminate it using Popen, it
             # will return `-N` as its exit code. As we normally get `N + 128` back, we
             # convert negative exit codes to the `N + 128` form.
             returncode = -nsjail.returncode + 128 if nsjail.returncode < 0 else nsjail.returncode
 
+            # Exclude user uploaded files from output
```

Comment:
What if the files changed?

Reply:
updated to upload user files if last modified time changes.
in  https://github.com/python-discord/snekbox/commit/36814de6f7a231c86b356ec27b649da5ffc85c44,